src/collection.ts

```diff
@@ -99,7 +99,10 @@ import {
 } from './utils';
 import { WriteConcern, WriteConcernOptions } from './write_concern';
 
-/** @public */
+/**
+ * @public
+ * @deprecated Will return the actual result document in 5.0
```

Comment:
I think we want to clarify here that this type will be removed and the `findOneAnd...` methods will return the result document directly

Reply:
Comment now more detailed about the deprecation.